contracts/contracts/oracle/BaseOETHOracleRouter.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/chainlink/AggregatorV3Interface.sol";
+import { OracleRouterBase } from "./OracleRouterBase.sol";
+import { StableMath } from "../utils/StableMath.sol";
+
+// @notice Oracle Router that denominates all prices in ETH for base network
+contract BaseOETHOracleRouter is OracleRouterBase {
+    using StableMath for uint256;
+
+    address public immutable aeroPriceFeed;
+
+    constructor(address _aeroPriceFeed) {
+        aeroPriceFeed = _aeroPriceFeed;
+    }
+
+    /**
+     * @notice Returns the total price in 18 digit units for a given asset.
+     *         This implementation does not (!) do range checks as the
+     *         parent OracleRouter does.
+     * @param asset address of the asset
+     * @return uint256 unit price for 1 asset unit, in 18 decimal fixed
+     */
+    function price(address asset)
```

Comment:
this function is still identical to the one in `OETHOracleRouter`. You can create a `OETHOracleRouterBase` and implement the `price` function in it. And then have `OETHOracleRouter` and `BaseOETHOracleRouter` inherit from it

Reply:
Refactored here: [826361f](https://github.com/OriginProtocol/origin-dollar/pull/2061/commits/826361ffd619e8c2f87bb7883d7fafb84f86c165)